chips/rp2040/src/pwm.rs

```diff
@@ -0,0 +1,1290 @@
+    }
+
+    // Force interrupt on the given channel
+    fn force_interrupt(&self, channel_number: ChannelNumber) {
+        let mask = self.registers.intf.read(CH::CH);
+        self.registers
+            .intf
+            .modify(CH::CH.val(mask | 1 << channel_number as u32));
+    }
+
+    // Unforce interrupt
+    fn unforce_interrupt(&self, channel_number: ChannelNumber) {
+        let mask = self.registers.intf.read(CH::CH);
+        self.registers
+            .intf
+            .modify(CH::CH.val(mask & !(1 << channel_number as u32)));
+    }
+
+    // Get interrupt status
+    fn get_interrupt_status(&self, channel_number: ChannelNumber) -> bool {
+        (self.registers.ints.read(CH::CH) & 1 << channel_number as u32) != 0
+    }
+
+    // Handle PWM interrupts
+    //
+    // This method should be called only inside crate::chip::Rp2040DefaultPeripherals::service_interrupt.
+    pub fn handle_interrupt(&self) {
+        for channel_number in CHANNEL_NUMBERS {
+            if self.get_interrupt_status(channel_number) {
+                self.interrupt_handler
+                    .map(|handler| handler.fired(channel_number));
+                self.clear_interrupt(channel_number);
+                self.unforce_interrupt(channel_number);
+            }
+        }
+    }
+
+    // Configure the given channel using the given configuration
+    fn configure_channel(&self, channel_number: ChannelNumber, config: &PwmChannelConfiguration) {
+        self.set_ph_correct(channel_number, config.ph_correct);
+        self.set_invert_polarity(channel_number, config.a_inv, config.b_inv);
+        self.set_div_mode(channel_number, config.divmode);
+        self.set_divider_int_frac(channel_number, config.int, config.frac);
+        self.set_compare_value_a(channel_number, config.cc_a);
+        self.set_compare_value_b(channel_number, config.cc_b);
+        self.set_top(channel_number, config.top);
+        self.set_enabled(channel_number, config.en);
+    }
+
+    /// Set an interrupt handler
+    ///
+    /// See [Interrupt]
+    pub fn set_interrupt_handler(&self, interrupt_handler: &'a dyn Interrupt) {
+        self.interrupt_handler.set(interrupt_handler);
+    }
+
+    // Initialize the struct
+    fn init(&self) {
+        let default_config: PwmChannelConfiguration = PwmChannelConfiguration::default();
+        for channel_number in CHANNEL_NUMBERS {
+            self.configure_channel(channel_number, &default_config);
+            self.set_counter(channel_number, 0);
+        }
+        self.registers.intr.write(CH::CH.val(0));
+    }
+
+    /// This method should be called when resolving dependencies for the
+    /// default peripherals. See [crate::chip::Rp2040DefaultPeripherals::resolve_dependencies]
+    pub fn set_clocks(&self, clocks: &'a clocks::Clocks) {
+        self.clocks.set(clocks);
+    }
+
+    // Given a channel number and a channel pin, return a struct that allows controlling it
```

Comment:
```suggestion
    /// Given a channel number and a channel pin, return a struct that allows controlling it
```

Reply:
Changed comments, although I thought only public functions should be included in the documentation.